packages/job-worker/src/blueprints/context/adlibActions.ts

```diff
@@ -636,4 +646,49 @@ export class ActionExecutionContext
 	async hackGetMediaObjectDuration(mediaId: string): Promise<number | undefined> {
 		return getMediaObjectDuration(this._context, mediaId, this.studioId)
 	}
+
+	async listPeripheralDevices(): Promise<(PeripheralDevicePublic & { actions: SubdeviceAction[] | undefined })[]> {
+		const parentDeviceIds = this._cache.PeripheralDevices.findAll(
+			(doc: PeripheralDevice) =>
+				doc.studioId === this._context.studioId && doc.type === PeripheralDeviceType.PLAYOUT
+		).map((doc) => doc._id)
+		if (parentDeviceIds.length === 0) {
+			throw new Error('No parent devices are configured')
+		}
+
+		const devices = await this._context.directCollections.PeripheralDevices.findFetch({
+			parentDeviceId: {
+				$in: parentDeviceIds,
+			},
+		})
+
+		return devices.map((d) => {
+			// Only expose a subset of the PeripheralDevice to the blueprints
+			return literal<Complete<PeripheralDevicePublic & { actions: SubdeviceAction[] | undefined }>>({
+				_id: d._id,
+				name: d.name,
+				deviceName: d.deviceName,
+				studioId: d.studioId,
+				category: d.category,
+				type: d.type,
+				subType: d.subType,
+				parentDeviceId: d.parentDeviceId,
+				created: d.created,
+				status: d.status,
+				settings: d.settings,
+				actions:
+					d.configManifest.subdeviceManifest &&
```

Comment:
could this be simplified to `actions: d.configManifest.subdeviceManifest?.[d.subType]?.actions,` ?

Reply:
TypeScript kept complaining when I tried this before but having tried it again it's now happy, I wonder if something wasn't built right